src/components/__tests__/input.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { render } from '@testing-library/react'
+import React from 'react'
+
+import { Input } from '@/components/input'
+
+describe('Input', () => {
+  it('should render an input with the `aria-label` attribute', () => {
+    const { queryByLabelText } = render(<Input aria-label="test input" />)
+
+    expect(queryByLabelText('test input')).toBeInTheDocument()
+    expect(queryByLabelText('test input')).toHaveAttribute('aria-label', 'test input')
+  })
+
+  it('should default to `type="text"`', () => {
+    const { queryByLabelText } = render(<Input aria-label="test input" />)
+
+    expect(queryByLabelText('test input')).toBeInTheDocument()
+    expect(queryByLabelText('test input')).toHaveAttribute('type', 'text')
+  })
+
+  it('should not have the `disabled` attribute and `aria-disabled="false"` if `loading` is false', () => {
+    const { queryByLabelText } = render(<Input aria-label="test input" />)
+
+    expect(queryByLabelText('test input')).toBeInTheDocument()
+    expect(queryByLabelText('test input')).not.toHaveAttribute('disabled')
+    expect(queryByLabelText('test input')).toHaveAttribute('aria-disabled', 'false')
+    expect(queryByLabelText('test input')).not.toBeDisabled()
+  })
+
+  it('should have the `disabled` and `aria-disabled="true"` attributes if `loading` is true', () => {
+    const { queryByLabelText } = render(<Input aria-label="test input" loading />)
+
+    expect(queryByLabelText('test input')).toBeInTheDocument()
+    expect(queryByLabelText('test input')).toHaveAttribute('disabled')
+    expect(queryByLabelText('test input')).toHaveAttribute('aria-disabled', 'true')
+    expect(queryByLabelText('test input')).toBeDisabled()
+  })
+
+  it('should have the `border-base` class by default', () => {
+    const { queryByLabelText } = render(<Input aria-label="test input" />)
+
+    expect(queryByLabelText('test input')).toBeInTheDocument()
+    expect(queryByLabelText('test input')).toHaveClass('border-base')
+  })
+
+  it('should have the `border-error` class when the `error` variant is passed', () => {
+    const { queryByLabelText } = render(<Input aria-label="test input" error />)
+
+    expect(queryByLabelText('test input')).toBeInTheDocument()
+    expect(queryByLabelText('test input')).toHaveClass('border-error')
+  })
```

Comment:
Let's also check if the error message is present in the DOM and if the input has the `aria-invalid` and `aria-labeledby` attributes.